api/server/request_controller.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gin-gonic/gin"
+	"github.com/iron-io/functions/api"
+	"github.com/iron-io/runner/common"
+)
+
+type RequestController interface {
+	AppName() string
```

Comment:
I agree with @treeder about passing down the populated App initially itself.

Usecase: suppose I have a middleware that based on the caller (identified from a HEADER) wants to change the limits (Memory/Timeout) of the container. This cannot be via the middleware now.